src/main/java/org/esbtools/gateway/resync/ResyncError.java

```diff
@@ -0,0 +1,13 @@
+package org.esbtools.gateway.resync;
+
+public final class ResyncError {
+
+    public static final String ALL_REQUIRED_VALUES_NOT_PRESENT = "One or more required values was not present";
+    public static final String SYSTEM_NOT_CONFIGURED = "The %s system is not configured properly";
+    public static final String PROBLEM_ENQUEUING = "There was a problem enqueuing the selected message: %s";
+
+    public static String withContext(String errorMessage, String variableValue) {
+        return String.format(errorMessage, variableValue);
+    }
```

Comment:
IMHO maybe a little cleaner way to do this would be to have different method for each "context" instead of static constants, then you can support multiple variables, and have some documentation about for a given error message what you're filling in:

```java
public static String allRequiredValuesNotPresent(String... requiredValueNames) {
  return "One or more of the required values were not present: " + Arrays.toString(requiredValueNames));
}

public static String systemNotConfigured(String systemName) {
  return String.format("The %s system is not configured properly", systemName);
}
```

etc.